src/select.tsx

```diff
@@ -31,25 +30,32 @@ interface CommonProps {
   loading?: boolean;
   loadingPlaceholder?: string;
   emptyPlaceholder?: string;
+  children?: JSX.Element;
 }
 
-type SelectReturn = ReturnType<typeof createSelect>;
+type SelectReturn<O,V> = ReturnType<typeof createSelect<O,V>>;
 
-type SelectProps = CreateSelectProps & Partial<CommonProps>;
+type SelectProps<O, V> = CreateSelectProps<O, V> & Partial<CommonProps<O,V>>;
 
-const SelectContext = createContext<SelectReturn>();
+function SelectContext<O,V>() {
+  return createContext<SelectReturn<O,V>>()
+};
```

Comment:
Note that this is a breaking change; it is now a function _returning_ a context. AFAIK only functions (and types) can have generics, so I had to wrap it in a fn.